scaleway/resource_function_namespace.go

```diff
@@ -0,0 +1,160 @@
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	function "github.com/scaleway/scaleway-sdk-go/api/function/v1beta1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayFunctionNamespace() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewayFunctionNamespaceCreate,
+		ReadContext:   resourceScalewayFunctionNamespaceRead,
+		UpdateContext: resourceScalewayFunctionNamespaceUpdate,
+		DeleteContext: resourceScalewayFunctionNamespaceDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Default: schema.DefaultTimeout(defaultFunctionNamespaceTimeout),
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				ForceNew:    true,
+				Optional:    true,
+				Description: "The name of the function namespace",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The description of the function namespace",
+			},
+			"environment_variables": {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: "The environment variables of the function namespace",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"registry_endpoint": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The endpoint reachable by docker",
+			},
+			"registry_namespace_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The ID of the registry namespace",
+			},
+			"region":          regionSchema(),
+			"organization_id": organizationIDSchema(),
+			"project_id":      projectIDSchema(),
+		},
+	}
+}
+
+func resourceScalewayFunctionNamespaceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api, region, err := functionAPIWithRegion(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	ns, err := api.CreateNamespace(&function.CreateNamespaceRequest{
+		Description:          expandStringPtr(d.Get("description").(string)),
+		EnvironmentVariables: expandMapStringStringPtr(d.Get("environment_variables")),
+		Name:                 expandOrGenerateString(d.Get("name").(string), "func-"),
+		ProjectID:            d.Get("project_id").(string),
+		Region:               region,
+	}, scw.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(newRegionalIDString(region, ns.ID))
+
+	return resourceScalewayFunctionNamespaceRead(ctx, d, meta)
+}
+
+func resourceScalewayFunctionNamespaceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api, region, id, err := functionAPIWithRegionAndID(meta, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	ns, err := api.GetNamespace(&function.GetNamespaceRequest{
```

Comment:
Please use [this](https://github.com/scaleway/scaleway-sdk-go/blob/master/api/function/v1beta1/function_helpers.go#L27) instead